product/roundhouse.tests/sqlsplitters/MySqlStatementSplitterSpecs.cs

```diff
@@ -0,0 +1,152 @@
+using System;
+using System.Collections.Generic;
+using NUnit.Framework;
+using roundhouse.databases.mysql.parser;
+using roundhouse.sqlsplitters;
+
+namespace roundhouse.tests.sqlsplitters
+{
+    public class MySqlStatementSplitterSpecs
+    {
+        public abstract class concern_for_StatementSplitter : TinySpec
+        {
+            public override void Context()
+            {
+                
+            }
+        }
+
+        [Concern(typeof(StatementSplitter))]
+        public class when_splitting_sql_statements : concern_for_StatementSplitter
+        {
+            public override void Because()
+            {
+
+            }
+
+            [Observation]
+            public void should_split_on_delimiters()
+            {
+                string script = "select * from test1;\nselect * from test2;\n";
+
+                Parser parser = new Parser(script);
+                List<ParsedStatement> statements = parser.Parse();
+                WriteStatements(statements);
+
+                TestContext.Out.WriteLine("Statements parsed: " + statements.Count);
+                Assert.IsTrue(statements[0].Value.Length > 10);
+                Assert.AreEqual(statements.Count, 2);
+            }
+
+            [Observation]
+            public void should_retain_comments()
+            {
+                string script = "/*\n *\n * Test Statement\n *\n */\nselect * from test1;\nselect * from test2;\n";
+
+                Parser parser = new Parser(script);
+                List<ParsedStatement> statements = parser.Parse();
+                WriteStatements(statements);
+
+                TestContext.Out.WriteLine("Statements parsed: " + statements.Count);
+                Assert.IsTrue(statements[0].Value.Length > 25);
+                Assert.AreEqual(statements.Count, 2);
+            }
+
+            [Observation]
+            public void should_honor_single_character_delimiter()
+            {
+                string script = "delimiter $\nselect * from test1$\nselect * from test2$\n";
+
+                Parser parser = new Parser(script);
+                List<ParsedStatement> statements = parser.Parse();
+                WriteStatements(statements);
+
+                TestContext.Out.WriteLine("Statements parsed: " + statements.Count);
+                Assert.IsTrue(statements[0].Value.Length > 10);
+                Assert.AreEqual(statements.Count, 2);
+            }
+
+            [Observation]
+            public void should_honor_tricky_single_character_delimiter()
+            {
+                string script = "delimiter Z\nselect * from test1Z\nselect * from test2Z\n";
+
+                Parser parser = new Parser(script);
+                List<ParsedStatement> statements = parser.Parse();
+                WriteStatements(statements);
+
+                TestContext.Out.WriteLine("Statements parsed: " + statements.Count);
+                Assert.IsTrue(statements[0].Value.Length > 10);
```

Comment:
Same here - I see that the resulting statement contains a trailing `\n`, even thought the delimiter `$` is before the `\n` in the original string. Is that expected? It would be great if the test expected the exact expected outcome, as it is much easier to understand the intent of the parsing - e.g. if the trailing newline should be included in the statement or not.

Could you please change the assertion accordingly, i.e:
 Assert.That(statements[0].Value.Contains("Test Statement"));
```suggestion
                Assert.AreEqual(statements[0].Value, "select * from test1\n");
```

(the above fails as of now)

or, if you didn't mean to have the newline in the statement,

 Assert.That(statements[0].Value.Contains("Test Statement"));
```suggestion
                Assert.AreEqual(statements[0].Value, "select * from test1");
```

Reply:
Each parsed statement is the text of the statement, minus the delimiter. For MySQL, anyway, it doesn't want to receive the actual delimiter and treats every statement we send as a complete statement.

When tokens are gathered into statements, every "EndOfLine" token results in the correct line ending for the environment being appended (Environment.NewLine). In this case the "$" is being left off (the delimiter isn't being sent) and then, I believe, the trailing line ending is being added because the next token is an EndOfLine.

I think I've changed all of these test cases to be more explicit.

Thank you!